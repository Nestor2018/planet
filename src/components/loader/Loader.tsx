import React from "react";
import {
  StyleSheet,
  ActivityIndicator,
  ActivityIndicatorProps,
} from "react-native";

interface LoaderProps extends ActivityIndicatorProps {}

const Loader: React.FC<LoaderProps> = ({
  color = "white",
  size = "large",
  style,
  ...rest
}) => (
  <ActivityIndicator
    color={color}
    size={size}
    style={[styles.loader, style]}
    {...rest}
    testID="loader"
    accessibilityLabel={rest.accessibilityLabel || "Loading..."}
  />
);

const styles = StyleSheet.create({
  loader: {
    margin: 20,
  },
});

export default Loader;
