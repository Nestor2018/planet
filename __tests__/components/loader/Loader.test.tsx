import React, { memo } from "react";
import { render } from "@testing-library/react-native";
import Loader from "../../../src/components/loader/Loader";

describe("Loader Component", () => {
  it("should render the Loader component", () => {
    const { getByTestId } = render(<Loader />);
    const loader = getByTestId("loader");

    expect(loader).toBeTruthy();
  });

  it("should apply custom color and size", () => {
    const { getByTestId } = render(<Loader color="red" size="small" />);
    const loader = getByTestId("loader");

    expect(loader.props.color).toBe("red");
    expect(loader.props.size).toBe("small");
  });

  it("should combine default styles with custom styles", () => {
    const customStyle = { margin: 50 };
    const { getByTestId } = render(<Loader style={customStyle} />);
    const loader = getByTestId("loader");

    expect(loader.props.style).toEqual(expect.arrayContaining([customStyle]));
  });

  it("matches snapshot", () => {
    const tree = render(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("is accessible with proper accessibility labels", () => {
    const { getByTestId } = render(<Loader accessibilityLabel="Loading..." />);
    const loader = getByTestId("loader");

    expect(loader.props.accessibilityLabel).toBe("Loading...");
  });

  it("should pass additional props to ActivityIndicator", () => {
    const customProps = { animating: false, hidesWhenStopped: true };
    const { getByTestId } = render(<Loader {...customProps} />);
    const loader = getByTestId("loader");

    expect(loader.props.animating).toBe(false);
    expect(loader.props.hidesWhenStopped).toBe(true);
  });
});
