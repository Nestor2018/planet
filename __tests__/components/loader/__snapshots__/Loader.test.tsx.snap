// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component matches snapshot 1`] = `
<ActivityIndicator
  accessibilityLabel="Loading..."
  color="white"
  size="large"
  style={
    [
      {
        "margin": 20,
      },
      undefined,
    ]
  }
  testID="loader"
/>
`;
